src/rebar_compiler_erl.erl

```diff
@@ -78,7 +78,7 @@ needed_files(Graph, FoundFiles, _, AppInfo) ->
 dependencies(Source, SourceDir, Dirs) ->
     case file:open(Source, [read]) of
         {ok, Fd} ->
-            Incls = parse_attrs(Fd, [], SourceDir),
+            Incls = lists:usort(parse_attrs(Fd, [], SourceDir)),
```

Comment:
this call should take place within `parse_attrs` to fully isolate the contract and make it so the function only returns non-duplicated values.

Reply:
I wasn't sure whether to do that or not, seeing as parse_attrs is recursive (Anyway, I've gone ahead and done this =) )